### DISCUSSION TITLE
Do I need to be root to build step-cli?

### DISCUSSION THREAD

**Discussion by aaribaud** [smallstep/certificates#695] [General] [answered]
Hello,
Following the developer docs (docs/local-development.md), I am hitting a problem: `make bootstrap` wants to install the `golangci-lint` binary in `/bin/`, which it obviously cannot do unless I were building as *root*, which I would have reservations about, to say the least.

Is this an issue with my setup, or does `make Bootstrap` really want to add an executable to the system's `/bin/` directory? And if so, is there a reason why `golangci-lint` cannot simply be installed in `${HOME}/.local/bin/`?

**Comment by aaribaud** [comment]
Answering myself: issue with my setup apparently. `go env GOPATH` returns the right value in my shell, but not in make.

[EDIT: unmarked as answer, because the issue *might* be with the documentation or makefiles of `step-cli`]

**Comment by aaribaud** [comment]
Right, so I double-checked. What I did is:
* Installed Debian 11 in a VM, with make, curl, git and golang available
* Verified that go is working by running a hello, world go program example
* Cloned step-cli as ~/go/src/github.com/smallstep/cli, as per the `doc/local-development.md` instructions
* Ran `make bootstrap`.
The result is:
> `install: cannot create regular file '/bin/golangci-lint': Permission denied`

Investigation shows that make/common.mk invokes `$(go env GOPATH)` and gets an empty string from it (whereas it gives the proper /home/*user*/go path when executed from shell or script) and prepends that empty string to "/bin/golangci-lint", accidentally resulting in a system path.

The problem *seems* to be that the makefile should invoke `$$(go env GOPATH)`, with an escaped `$`, as this appears to be a bash construct, not a Make one. However, I guess if that were the issue, I would not be the only one having it, yet I found no other discussion about this.

Anyone have any idea?

**Comment by aaribaud** [comment, marked as answer]
Tested replacing `$(go env GOPATH)` with `$$(go env GOPATH)`, `make bootstrap` now runs fine. Once I've run all tests suggested in `doc/local-development.md`, I'll submit a PR for this and see if anyone screams about it.

**Reply by maraino** [reply]
Yes, it should be $$, but I've responded in https://github.com/smallstep/cli/issues/535, we might want to make sure that this will work on CI/CD.
